tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -211,6 +220,41 @@ static void printBufferedNotifications(bool syncWithService = true) {
   });
 }
 
+/// A simple configurable FileSystemProvider, useful for tests that exercise
+/// the FileSystemProvider code.
+class TestFileSystemProvider : public SourceKit::FileSystemProvider {
```

Comment:
Providing this as part of sourcekitd-test won't work when running on macOS using XPC, since it will be in the wrong process.  It should be fine to bake this file system into the service itself, although maybe we should call it something other than "testvfs" in that case.

Reply:
I'd like to keep the sourcekitd layer.  What do you think of my suggestion to move the file system provider down into SwiftLangSupport and resolve the name+options at that layer?  We could easily provide default arguments to make unit tests cleaner.